src/reportGenerator.js

```diff
@@ -39,17 +39,21 @@ class ReportGenerator {
 	}
 
 	/**
-	 * Returns the stylesheet to be requireed in the test report.
+	 * Returns the stylesheet to be required in the test report.
 	 * If styleOverridePath is not defined, it will return the defined theme file.
 	 * @return {Promise}
 	 */
 	getStylesheetContent() {
 		const pathToStylesheet = this.config.getStylesheetFilepath();
+		const useCssFile = this.config.shouldUseCssFile();
 		return new Promise((resolve, reject) => {
 			fs.readFile(pathToStylesheet, 'utf8', (err, content) => {
 				if (err) {
 					return reject(new Error(`Could not locate the stylesheet: '${pathToStylesheet}': ${err}`));
 				}
+				if (useCssFile) {
```

Comment:
Move the "shouldUseCssFile"-check along with the fetching of the stylesheet path into the `generate` method instead - to avoid confusion (as this method is explicitly named "getStylesheetContent"). There you would check whether to use this method or simply pass the path to the `renderHtmlReport`.